### PR TITLE
Introduce gps and bte ios targets

### PR DESCRIPTION
**Pull request by johnschoeman**
Why:
We would like to introduce bluetooth to the application for bluetooth
based contact tracing. Given the privacy concerns of having both
location and bluetooth data available to the same application, google
and apple do not allow for both location and bluetooth to be available
at the same time.

Hence in order to service both location and bluetooth
we will need to provide separate apps with separate info.plists and
bundle identifiers. At the same time we would like to share a much
application logic between the two applications as to reduce the
maintenance and organizational overhead. Ideally as close to 100% of the
javascript layer is common and for the ios layer we only need to change
the info.plists.

This commit:

Introduces two build targets, GPS and BTE, to the xcode project which
are setup to read from the corresponding groups for config and info.plist
A COVIDSafePathsConfig file was introduced to expose a
'getTracingStrategy' function to the react native side so that the js
runtime can determine which tracing strategy the bundle will be using.
This will allow us flexibility to programmatically determine which
tracing strategy is being used either within a given react component or
to conditional render components altogether. This also allows use to
maximize the amount of shared native code.

Three schemes where added for each target, a Development, a Staging, and
a Production.

To build the corresponding targets via the react native cli, you can
pass the --scheme flag with the corresponding scheme, e.g:

```
react-native run-ios --scheme "GPS_Development"
react-native run-ios --scheme "BTE_Development"
```

We updated the `yarn run-ios` script in package.json to two scripts for
for the gps based build and the bte based build, `yarn run-ios-bte` and
`yarn run-ios-gps` respectively

Note that theres a fair amount of location information storing logic
currently in AppDelegate, a future commit might want to refactor this so
that the gps specific code is only built in the GPS target builds.

### GPS
Screen Shot 2020-05-20 at 4 19 14 PM

### BTE
Screen Shot 2020-05-20 at 4 17 47 PM 

### Testing

Verify you can run both versions of the app.
- reinstall your pods, clean build folder, remove derived data, etc.
- run `yarn run-ios-gps`
- run `yarn run-ios-bte`

### CODE REVIEW

File: .gitignore
Comment (johnschoeman) [inline .gitignore:47]: this got missed in a previous commit.

File: ios/GPS/Info.plist
Comment (johnschoeman) [inline ios/GPS/Info.plist]: This shouldnt be changed.

File: ios/COVIDSafePaths/Base.lproj/LaunchScreen.xib
Comment (johnschoeman) [inline ios/COVIDSafePaths/Base.lproj/LaunchScreen.xib]: we shouldnt need to change the launch screen in this one.

**Comment by johnschoeman** [comment]
the next increment to introduce to develop from the bte branch would be the fastlane work.

**Review by mattThousand — DISMISSED** [review]

**Comment by mattThousand** [comment]
LGTM once those unit tests are green 🍏 😄

**Review by tstirrat — DISMISSED** [review]
Would like to get a weigh in from @efalkner in regards to fastlane/workflows too before we merge.

**Comment by johnschoeman** [comment]
> Would like to get a weigh in from @efalkner in regards to fastlane/workflows too before we merge.

@efalkner and i actually paired on the fastlane/workflows on a previous thread! Im currently working on getting a pr specifically for those up, for ease of review with the rest of the team.  That said, this pr wont affect the current github workflow at all and the bte stuff wont auto build on ci.

File: .github/workflows/test.yml
Comment (johnschoeman) [inline .github/workflows/test.yml:109]: Ended up needing to split the test targets as well. COVIDSafePathsTests became GPSTests

**Comment by johnschoeman** [comment]
ping @mattThousand, @tstirrat , looks like my for push reset your reviews :/ if i could trouble you for another review. Ive add changes to fix up the issue with the xcode test suite. with a comment in the code.

**Review by mattThousand — APPROVED** [review]

**Pull request merged**
